server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java

```diff
@@ -62,59 +72,90 @@ public String getName() {
     }
 
     static final String BLOBSTORE_IMPLEMENTATION_PROPERTY = "implementation";
+    static final String CACHE_ENABLE_PROPERTY = "cache.enable";
+
+    public static BlobStoreConfiguration parse(org.apache.james.server.core.configuration.Configuration configuration) throws ConfigurationException {
+        PropertiesProvider propertiesProvider = new PropertiesProvider(new FileSystemImpl(configuration.directories()), configuration);
+
+        return parse(propertiesProvider);
+    }
+
+    public static BlobStoreConfiguration parse(PropertiesProvider propertiesProvider) throws ConfigurationException {
+        try {
+            Configuration configuration = propertiesProvider.getConfigurations(ConfigurationComponent.NAMES);
+            return BlobStoreConfiguration.from(configuration);
+        } catch (FileNotFoundException e) {
+            LOGGER.warn("Could not find " + ConfigurationComponent.NAME + " configuration file, using cassandra blobstore as the default");
+            return BlobStoreConfiguration.cassandra();
+        }
+    }
 
-    static BlobStoreChoosingConfiguration from(Configuration configuration) {
+    static BlobStoreConfiguration from(Configuration configuration) {
         BlobStoreImplName blobStoreImplName = Optional.ofNullable(configuration.getString(BLOBSTORE_IMPLEMENTATION_PROPERTY))
             .filter(StringUtils::isNotBlank)
             .map(StringUtils::trim)
             .map(BlobStoreImplName::from)
             .orElseThrow(() -> new IllegalStateException(String.format("%s property is missing please use one of " +
                 "supported values in: %s", BLOBSTORE_IMPLEMENTATION_PROPERTY, BlobStoreImplName.supportedImplNames())));
 
-        return new BlobStoreChoosingConfiguration(blobStoreImplName);
+        boolean cacheEnabled = configuration.getBoolean(CACHE_ENABLE_PROPERTY, false);
+
+        return new BlobStoreConfiguration(blobStoreImplName, cacheEnabled);
+    }
+
+    public static BlobStoreConfiguration cassandra() {
+        return new BlobStoreConfiguration(BlobStoreImplName.CASSANDRA, !CACHE_ENABLED);
     }
 
-    public static BlobStoreChoosingConfiguration cassandra() {
-        return new BlobStoreChoosingConfiguration(BlobStoreImplName.CASSANDRA);
+    public static BlobStoreConfiguration objectStorage() {
+        return new BlobStoreConfiguration(BlobStoreImplName.OBJECTSTORAGE, !CACHE_ENABLED);
     }
 
-    public static BlobStoreChoosingConfiguration objectStorage() {
-        return new BlobStoreChoosingConfiguration(BlobStoreImplName.OBJECTSTORAGE);
+    public static BlobStoreConfiguration cachingEnabled() {
```

Comment:
it's implicitly `objectStorage` but it's almost impossible to guess without reading the code

Reply:
Lets call it objectStorageWithCachingEnabled module

Reply:
what about having a fluent choice instead?